source/common/upstream/logical_dns_cluster.cc

```diff
@@ -68,47 +69,32 @@ void LogicalDnsCluster::startResolve() {
   ENVOY_LOG(debug, "starting async DNS resolution for {}", dns_address);
   info_->stats().update_attempt_.inc();
 
-  active_dns_query_ = dns_resolver_->resolve(
-      dns_address, dns_lookup_family_,
-      [this,
-       dns_address](std::list<Network::Address::InstanceConstSharedPtr>&& address_list) -> void {
-        active_dns_query_ = nullptr;
-        ENVOY_LOG(debug, "async DNS resolution complete for {}", dns_address);
-        info_->stats().update_success_.inc();
-
-        if (!address_list.empty()) {
-          // TODO(mattklein123): Move port handling into the DNS interface.
-          ASSERT(address_list.front() != nullptr);
-          Network::Address::InstanceConstSharedPtr new_address =
-              Network::Utility::getAddressWithPort(*address_list.front(),
-                                                   Network::Utility::portFromTcpUrl(dns_url_));
-          if (!logical_host_) {
-            logical_host_.reset(
-                new LogicalHost(info_, hostname_, new_address, localityLbEndpoint(), lbEndpoint()));
-
-            const auto& locality_lb_endpoint = localityLbEndpoint();
-            PriorityStateManager priority_state_manager(*this, local_info_, nullptr);
-            priority_state_manager.initializePriorityFor(locality_lb_endpoint);
-            priority_state_manager.registerHostForPriority(logical_host_, locality_lb_endpoint);
-
-            const uint32_t priority = locality_lb_endpoint.priority();
-            priority_state_manager.updateClusterPrioritySet(
-                priority, std::move(priority_state_manager.priorityState()[priority].first),
-                absl::nullopt, absl::nullopt, absl::nullopt);
-          }
-
-          if (!current_resolved_address_ || !(*new_address == *current_resolved_address_)) {
-            current_resolved_address_ = new_address;
-
-            // Make sure that we have an updated address for admin display, health
-            // checking, and creating real host connections.
-            logical_host_->setNewAddress(new_address, lbEndpoint());
-          }
-        }
-
-        onPreInitComplete();
-        resolve_timer_->enableTimer(dns_refresh_rate_ms_);
-      });
+  if (srv_) {
```

Comment:
It seems unfortunate that we need to make strict/logical DNS clusters aware of the detail of SRV. Ideally, the resolver extension mechanism could make this opaque. The main issue we have today is the resolver interface is synchronous (probably why you had the `Wait()` above). If you can fix this to be more asynchronous, we could just make SRV handling a first class address resolver plugin and not have to complicate the different cluster types.